posthog/temporal/data_imports/sources/stripe/stripe.py

```diff
@@ -46,119 +48,165 @@ class StripeNestedResource:
     params: dict[str, Any] = dataclasses.field(default_factory=dict)
 
 
-def stripe_source(
+@dataclasses.dataclass
+class StripeResumeConfig:
+    starting_after: str
+
+
+def get_rows(
     api_key: str,
-    account_id: Optional[str],
     endpoint: str,
+    account_id: Optional[str],
     db_incremental_field_last_value: Optional[Any],
     db_incremental_field_earliest_value: Optional[Any],
     logger: FilteringBoundLogger,
+    resumable_source_manager: ResumableSourceManager[StripeResumeConfig],
```

Comment:
Not a blocker but general comment that it seems like adding this is yet another piece of bespoke logic that needs to be accounted for when creating a source. It'd be ideal if we could structure sources in a way where: [source does extraction logic] -> [middleware that handles sync state/resumability/etc] -> [pipeline writer]

Reply:
Yeah, I hear you - the sources are doing a little too much. This is on my list for when we move to a consumer/producer architecture